src/upgrade/assessmentManager.ts

```diff
@@ -184,11 +163,44 @@ async function getWorkspaceIssues(workspaceFolderUri: string): Promise<UpgradeIs
             operationName: "java.dependency.assessmentManager.getWorkspaceIssues",
         });
         return [];
-    }).reduce((a, b) => [...a, ...b]);
+    }).flat();
 
     return workspaceIssues;
 }
 
+async function getAllDependencies(projectNode: INodeData): Promise<PackageDescription[]> {
+    const projectStructureData = await Jdtls.getPackageData({ kind: NodeKind.Project, projectUri: projectNode.uri });
+    const packageContainers = projectStructureData.filter(x => x.kind === NodeKind.Container);
+
+    const allPackages = await Promise.allSettled(
+        packageContainers.map(async (packageContainer) => {
+            const packageNodes = await Jdtls.getPackageData({
+                kind: NodeKind.Container,
+                projectUri: projectNode.uri,
+                path: packageContainer.path,
+            });
+            return packageNodes.map(packageNodeToDescription).filter((x): x is PackageDescription => Boolean(x));
+        })
+    );
+
+    return allPackages
+        .map(x => {
+            if (x.status === "fulfilled") {
+                return x.value;
+            }
+            sendInfo("", {
```

Comment:
I won't suggest send telemetry here in map. In case of big project that contains hundreds/thousands of packages, you may send big amount of telemetry event in a very short time, and causes telemetry data loss. 

I would suggest set telemetry properties here and send only one event in the end.